src/plugins/intel_npu/src/utils/include/intel_npu/utils/zero/zero_utils.hpp

```diff
@@ -11,24 +11,39 @@
 #include "intel_npu/al/config/runtime.hpp"
 #include "intel_npu/utils/logger/logger.hpp"
 #include "intel_npu/utils/zero/zero_result.hpp"
+#include "zero_types.hpp"
 
 namespace intel_npu {
 
 namespace zeroUtils {
 
-static inline void throwOnFail(const std::string& step, const ze_result_t result) {
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("L0 ",
-                       step,
-                       " result: ",
-                       ze_result_to_string(result),
-                       ", code 0x",
-                       std::hex,
-                       uint64_t(result),
-                       " - ",
-                       ze_result_to_description(result));
+#define THROW_ON_FAIL_FOR_LEVELZERO_EXT(step, result, graph_ddi_table_ext)              \
+    if (ZE_RESULT_SUCCESS != result) {                                                  \
+        OPENVINO_THROW("L0 ",                                                           \
+                       step,                                                            \
+                       " result: ",                                                     \
+                       ze_result_to_string(result),                                     \
+                       ", code 0x",                                                     \
+                       std::hex,                                                        \
+                       uint64_t(result),                                                \
+                       " - ",                                                           \
+                       ze_result_to_description(result),                                \
+                       " . ",                                                           \
+                       intel_npu::zeroUtils::getLatestBuildError(graph_ddi_table_ext)); \
+    }
+
+#define THROW_ON_FAIL_FOR_LEVELZERO(step, result)         \
+    if (ZE_RESULT_SUCCESS != result) {                    \
+        OPENVINO_THROW("L0 ",                             \
+                       step,                              \
+                       " result: ",                       \
+                       ze_result_to_string(result),       \
+                       ", code 0x",                       \
+                       std::hex,                          \
+                       uint64_t(result),                  \
+                       " - ",                             \
+                       ze_result_to_description(result)); \
     }
-}
 
 static inline void throwOnFail(const std::string& step, const ze_result_t result, const std::string& hintOnError) {
```

Comment:
Is this still needed? Can we keep the same design for all 3 functions?

Reply:
Hi, This function is not actually used. It is not known whether it will be used in the future, so the original version is retained. I can change it to a macro definition if we need to.

Reply:
I think maybe we can remove the function because it is not actually used.